src/Appwrite/Migration/Version/V20.php

```diff
@@ -45,19 +44,228 @@ public function execute(): void
+        };
+
+        // Support database array type migration (user collections)
+        $attributes = $this->projectDB->find('attributes', [
+            Query::equal('array', [true]),
+        ]);
+
+        foreach ($attributes as $attribute) {
+            $indexes = $this->projectDB->find('indexes', [
+                Query::equal('databaseInternalId', [$attribute['databaseInternalId']]),
+                Query::equal('collectionInternalId', [$attribute['collectionInternalId']]),
+            ]);
+            $foundIndex = 0;
+            foreach ($indexes as $index) {
+                if (in_array($attribute['key'], $index['attributes'])) {
+                    $this->projectDB->deleteIndex($index['collectionId'], $index['_uid']);
+                    $foundIndex = 1;
+                }
+            }
+            if ($foundIndex === 1) {
+                $this->projectDB->updateAttribute($attribute['collectionInternalId'], $attribute['key'], Database::VAR_STRING);
+            }
+        }
+
+        $collections = $this->collections[$collectionType];
+        foreach ($collections as $collection) {
+            $id = $collection['$id'];
+
+            Console::log("Migrating Collection \"{$id}\"");
+
+            $this->projectDB->setNamespace("_$internalProjectId");
+
+            // Support database array type migration
+             $foundIndex = 0;
+            foreach ($collection['attributes'] ?? [] as $attribute) {
+                if ($attribute['array'] === true) {
+                    foreach ($collection['indexes'] ?? [] as $index) {
+                        if (in_array($attribute['$id'], $index['attributes'])) {
+                            $this->projectDB->deleteIndex($id, $index['$id']);
+                              $foundIndex = 1;
+                        }
+                    }
+                    if ($foundIndex === 1) {
+                        $this->projectDB->updateAttribute($id, $attribute['$id'], Database::VAR_STRING);
+                    }
+                }
+            }
+
+            switch ($id) {
+                case '_metadata':
+                    $this->createCollection('providers');
+                    $this->createCollection('messages');
+                    $this->createCollection('topics');
+                    $this->createCollection('subscribers');
+                    $this->createCollection('targets');
+                    $this->createCollection('challenges');
+
+                    break;
+                case 'stats':
+                    try {
+                        /**
+                         * Delete 'type' attribute
+                         */
+                        $this->projectDB->deleteAttribute($id, 'type');
+                        /**
+                         * Alter `signed`  internal type on `value` attr
+                         */
+                        $this->projectDB->updateAttribute($id, 'value', null, null, null, null, true);
```

Comment:
Let's use named parameters here so we can skip the nulls and it's more clear what's changing